### ISSUE TITLE
Swupd slow downloading

### ISSUE THREAD

**Issue by Wello6143** [clearlinux/swupd-client#1341]
Hi,
I had jumped over to Clear Linux. I've just found out that swupd only used ~25% of the bandwidth. Because when I download the ISO file from alternative CDN, it uses the whole bandwidth (5.7MB/s), but when I download the packages bundle with swupd the speed is just ~1.7MB/s.
I do know about --max-parallel-download but still happen the same.
So how do I resolve this issue? 
Thanks.

**Comment by Wello6143** [comment]
I've tried: `download.clearlinux.org`, `cdn.download.clearlinux.org`, `cdn-alt.download.clearlinux.org` and they're all the same because I found out that all of these 3 actually the same one, same CDN deploy from one server but alternative subdomain.

**Comment by Wello6143** [comment]
Note: I do know about `swupd mirror` since I had read all the docs before rolled to Clear Linux

**Comment by otaviobp** [comment]
Nice. I'll set this as a feature request then.
We had an effort on getting download speed better in the past, but it looks like something we need to readdress. Thank you for the feedback.

**Comment by Wello6143** [comment]
I suggest to add something like parallel threads downloading for it.

**Comment by Ralim** [comment]
For what is worth as another data-point; 
I'm down in Australia but on a 100/40mbps fiber connection.
The highest `swupd` download speed I've recorded in the last few months is 100KB/sec, with an average of around 50KB/sec. This is very tediously slow.

Would love a way to run threads in parallel for the same file as the `bundle-add` often bogs down when it hits a 100MB+ bundle.

**Comment by Wello6143** [comment]
It's look like getting hard for devs to solving the `swupd`'s problems in milestones since the spread of COVID-19.

**Comment by sn99** [comment]
Any progress on this ?

**Comment by Wello6143** [comment]
> Any progress on this ?

Pretty sure nope. I have waited so long so I chose Fedora anyways.

**Comment by Ralim** [comment]
Don't think anything has changed much.
Feels like for myself at least, that files are not staying cached for longer than an hour or so.

If I update two machines back to back, the second one is nice and fast (200mbit or so), by the first one is painfully slow.

If I do one machine in the morning and one in the afternoon both are very slow.

**Comment by lars-whta** [comment]
It is still incredible slow, swupd can average around a few kbit/s while other CDNs can average around 300-500 Mbit/s on the same machine under the same conditions.

**Comment by FrozzDay** [comment]
I tried to compare the speed of the repo server with debian's.
Download speeds on Debian repos reach 60MBps while Clear Linux is only about 4.8MBps.

Test using the OS files available in the repo.

**Comment by wello6143** [comment]
Wow, it's been 2 years already and I forgot about this issue!
I currently am using Arch Linux, it was a great journey along the way.

**Comment by jmakov** [comment]
Yup, still the case. Would be great if the mirrors would be auto selected.

**Comment by FrozzDay** [comment]
better leave it auto update, as cached on akaimai

**Comment by Keltere** [comment]
I can confirm also that the speed was so low after 2h still was downloading packages.
I ended up ditching this distro because of this low mirror speed. Unfortunately.
Hope for a fix in the future, i was really eager to try it on my nas.

**Comment by git-Hmmm** [comment]
will this be ever addressed?

**Comment by fenrus75** [comment]
we are constantly keeping an eye out but it seems to greatly depend on
where in the world you are for the CDN to decide to be slow or fast

**Comment by whytf** [comment]
> For what is worth as another data-point; I'm down in Australia but on a 100/40mbps fiber connection. The highest `swupd` download speed I've recorded in the last few months is 100KB/sec, with an average of around 50KB/sec. This is very tediously slow.
> 
> Would love a way to run threads in parallel for the same file as the `bundle-add` often bogs down when it hits a 100MB+ bundle.

No worries i get about 130kb/s on load balanced dual wan 2Gbps/1.3Gbps in Europe
#1614
